File: app/src/main/java/io/mgba/Data/Remote/RetrofitClient.java
package io.mgba.Data.Remote;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {
    private Retrofit retrofit = null;

    public Retrofit getClient(String baseUrl) {
        if (retrofit == null) {
            synchronized (this){
                if(retrofit == null)
                    retrofit = new Retrofit.Builder()
                            .baseUrl(baseUrl)
                            .addConverterFactory(GsonConverterFactory.create())
                            .build();
            }
        }

        return retrofit;
    }
}

File: app/src/main/java/io/mgba/data/remote/RetrofitClient.kt
package io.mgba.data.remote

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitClient {
    private var retrofit: Retrofit? = null

    fun getClient(baseUrl: String): Retrofit {
        if (retrofit == null) {
            synchronized(this) {
                if (retrofit == null)
                    retrofit = Retrofit.Builder()
                            .baseUrl(baseUrl)
                            .addConverterFactory(GsonConverterFactory.create())
                            .build()
            }
        }

        return retrofit!!
    }
}
